engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -618,45 +608,50 @@ private void renderSimpleBlendMaterials() {
         PerformanceMonitor.endActivity();
     }
 
-    private void renderChunk(RenderableChunk chunk, ChunkMesh.RenderPhase phase, Camera camera, ChunkRenderMode mode) {
-        if (chunk.hasMesh()) {
-            final Vector3f cameraPosition = camera.getPosition();
-            final Vector3f chunkPosition = chunk.getPosition().toVector3f();
-            final Vector3f chunkPositionRelativeToCamera =
-                    new Vector3f(chunkPosition.x * ChunkConstants.SIZE_X - cameraPosition.x,
-                            chunkPosition.y * ChunkConstants.SIZE_Y - cameraPosition.y,
-                            chunkPosition.z * ChunkConstants.SIZE_Z - cameraPosition.z);
-
-            if (mode == ChunkRenderMode.DEFAULT || mode == ChunkRenderMode.REFLECTION) {
-                if (phase == ChunkMesh.RenderPhase.REFRACTIVE) {
-                    chunkShader.activateFeature(ShaderProgramFeature.FEATURE_REFRACTIVE_PASS);
-                } else if (phase == ChunkMesh.RenderPhase.ALPHA_REJECT) {
-                    chunkShader.activateFeature(ShaderProgramFeature.FEATURE_ALPHA_REJECT);
-                }
+    private void renderChunk(PriorityQueue<RenderableChunk> chunks, ChunkMesh.RenderPhase phase, Camera camera, ChunkRenderMode mode) {
+        final Vector3f cameraPosition = camera.getPosition();
+        if (mode == ChunkRenderMode.DEFAULT || mode == ChunkRenderMode.REFLECTION) {
+            if (phase == ChunkMesh.RenderPhase.REFRACTIVE) {
+                chunkShader.activateFeature(ShaderProgramFeature.FEATURE_REFRACTIVE_PASS);
+            } else if (phase == ChunkMesh.RenderPhase.ALPHA_REJECT) {
+                chunkShader.activateFeature(ShaderProgramFeature.FEATURE_ALPHA_REJECT);
+            }
 
-                chunkShader.setFloat3("chunkPositionWorld", chunkPosition.x * ChunkConstants.SIZE_X,
-                        chunkPosition.y * ChunkConstants.SIZE_Y,
-                        chunkPosition.z * ChunkConstants.SIZE_Z);
-                chunkShader.setFloat("animated", chunk.isAnimated() ? 1.0f : 0.0f);
+            if (mode == ChunkRenderMode.REFLECTION) {
+                chunkShader.setFloat("clip", camera.getClipHeight(), true);
+            } else {
+                chunkShader.setFloat("clip", 0.0f, true);
+            }
 
-                if (mode == ChunkRenderMode.REFLECTION) {
-                    chunkShader.setFloat("clip", camera.getClipHeight());
-                } else {
-                    chunkShader.setFloat("clip", 0.0f);
-                }
+            chunkShader.enable();
 
-                chunkShader.enable();
+        } else if (mode == ChunkRenderMode.SHADOW_MAP) {
+            shadowMapShader.enable();
 
-            } else if (mode == ChunkRenderMode.SHADOW_MAP) {
-                shadowMapShader.enable();
+        } else if (mode == ChunkRenderMode.Z_PRE_PASS) {
+            context.get(ShaderManager.class).disableShader();
```

Comment:
Minor issue from the original code, so not your fault @Josharias: any chance you can change this to obtain the ShaderManager on class initialization? Look in the method initRenderingSupport() where the ShaderManager is already obtained once from the context. Let's make that fill a private variable so we don't have to fetch it from the context every time.